File: 240424/정보에 따른 숫자 2/number-based-on-information-2.py
T, a, b = map(int, input().split())
S, N = [], []
poses = [0] * b
for _ in range(T):
    alphabet, pos = input().split()
    
    if alphabet == 'S':
        S.append(int(pos))
    elif alphabet == 'N':
        N.append(int(pos))

S.sort()
N.sort()
s_idx, n_idx, nearest_cnts = 0, 0, 0

for cur_pos in range(a, b + 1):
    while s_idx < len(S) - 1 and abs(cur_pos - S[s_idx]) > abs(cur_pos - S[s_idx + 1]):
        s_idx += 1

    s_dst = abs(cur_pos - S[s_idx])
    
    while n_idx < len(N) - 1 and abs(cur_pos - N[n_idx]) > abs(cur_pos - N[n_idx + 1]):
        n_idx += 1
    
    n_dst = abs(cur_pos - N[n_idx])

    if s_dst <= n_dst:
        nearest_cnts += 1

print(nearest_cnts)

File: 260329/정보에 따른 수 2/number-based-on-information-2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int T, a, b;
    cin >> T >> a >> b;

    vector<int> S, N;
    S.reserve(T);
    N.reserve(T);

    for (int i = 0; i < T; i++) 
    {
        char alphabet;
        int pos;
        cin >> alphabet >> pos;
        if (alphabet == 'S') S.push_back(pos);
        else if (alphabet == 'N') N.push_back(pos);
    }

    sort(S.begin(), S.end());
    sort(N.begin(), N.end());

    int s_idx = 0, n_idx = 0;
    long long nearest_cnts = 0;

    for (int cur_pos = a; cur_pos <= b; cur_pos++) 
    {
        while (s_idx < (int)S.size() - 1 &&
               llabs((long long)cur_pos - S[s_idx]) > llabs((long long)cur_pos - S[s_idx + 1])) 
        {
            s_idx++;
        }
        long long s_dst = llabs((long long)cur_pos - S[s_idx]);

        while (n_idx < (int)N.size() - 1 &&
               llabs((long long)cur_pos - N[n_idx]) > llabs((long long)cur_pos - N[n_idx + 1])) 
        {
            n_idx++;
        }
        long long n_dst = llabs((long long)cur_pos - N[n_idx]);

        if (s_dst <= n_dst) nearest_cnts++;
    }

    cout << nearest_cnts << "\n";
    return 0;
}